providers/common.rb

```diff
@@ -0,0 +1,56 @@
+action :install do
+
+  if new_resource.create_user
+    group new_resource.group do
+      action :create
+    end
+    user new_resource.owner do
+      action :create
+      gid new_resource.group
+    end
+  end
+
+  directory new_resource.path do
+    action :create
+    owner new_resource.owner
+    group new_resource.group
+  end
+
+  # Update the code.
+  git new_resource.path do
```

Comment:
I depend on the git cookbook in my metadata. I thought that installed git for me, but apparently not.